Taxi/Taxi/Extension/StringExtension.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  StringExtension.swift
+//  Taxi
+//
+//  Created by sanghyo on 2022/06/15.
+//
+
+import Foundation
+
+extension String {
+    static func isValidNickname(nickname: String) -> FieldState {
```

Comment:
```suggestion
    var isValidNickname: FieldState {
```

computed property로 바꾸면, 사용할 때도 직관적이고 parameter도 생략가능할지두 .. !!!!

```swift
let nickname = "lemon"

if nickname. isValidNickname { ✨ 
    print(nickname)
}
```

Reply:
string extension으로 만들다보니, 모든 string에 대해 isValidNickname method가 뜨는게 옳은것인가? 라는 고민에 그냥 static으로 작성했었습니다.
하지만 이렇게 사용하는게 더 간단하고 좋은 것 같아서 배우고 반영하겠습니당.
파라미터 없는건 computed property로!!